portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,18 +37,26 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long maxComments = Long.parseLong(request.getParameter("maxComments"));
+
+    // a value of -1 indicates that ALL comments are wanted
+    if (maxComments == -1) maxComments = Long.MAX_VALUE;
+
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
     List<Comment> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
+    Iterator<Entity> iterator = results.asIterable().iterator();
+    while ( iterator.hasNext() && comments.size() < maxComments ) {
+      Entity entity = iterator.next();
```

Comment:
alternatively you could leave the for loop and just add an if statement to break out of the loop.

Reply:
I've just reverted this change now 👍 https://github.com/thomasogara/step-repository/pull/28/commits/46a12ea2c1b8144cdbe86880ebc6742293fa516f